infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -600,8 +611,42 @@ public void accept(ProbeResult probeResult) {
     }
   }
 
+  /** Listens container's events and terminates workspace if unrecoverable event occurs. */
+  public class UnrecoverableEventHanler implements ContainerEventHandler {
+    private List<String> events;
+
+    public UnrecoverableEventHanler() {
+      this.events =
+          StringUtils.isNotBlank(unrecoverableEvents)
+              ? Arrays.asList(unrecoverableEvents.split(","))
+              : Collections.EMPTY_LIST;
+    }
+
+    @Override
+    public void handle(ContainerEvent event) {
+      if (isUnrecoverable(event)) {
+        String reason = event.getReason();
+        String message = event.getMessage();
+        String podName = event.getPodName();
+        try {
+          internalStop(emptyMap());
+        } catch (InfrastructureException e) {
+          LOG.error("Unrecoverable event occured {}, {}, {}", reason, message, podName);
```

Comment:
It seems that without workspace ID it won't be possible to match the incident with a workspace/user. I think we need WSID her.

Reply:
ok, will also add workspace id